### ISSUE TITLE
Add support for: Kingston (London)

### ISSUE THREAD

**Issue by ManicGTI** [mampfes/hacs_waste_collection_schedule#384]
Would there be any chance Kingston, London could be added?

https://kingston-self.achieveservice.com/service/in_my_area?displaymode=collections

Example postcode: https://kingston-self.achieveservice.com/service/In_my_Area_Results?uprn=100021772910&displaymode=collections&altVal=

I had a go myself but failed miserably, it's beyond my knowledge level as there is no easy calender to extract.

Thank you!

**Comment by DevilasNZ** [comment]
I might be able to have a crack at this but would highly recommend learning this stuff yourself; I didn't have any real experience reverse engineering APIs until learning and adding support for my area.

This video from Postman was a big help for me: https://www.youtube.com/watch?v=2VKVPB5USgU
Postman also has a very cool feature which converts your request to code (in our case it's Python-Requests). Basically it's a matter of finding the request that includes the information you want and translating that to a more readable format. My case was a bit tricky with keeping a live session but hopefully this should be a bit more straight forward.

Good luck!

**Comment by skipishere** [comment]
Looks like they made this one difficult to reverse engineer
One of the request that worked looked like this
```
https://kingston-self.achieveservice.com/apibroker/runLookup?id=601a61f9a3188&repeat_against=&noRetry=true&getOnlyTokens=undefined&log_id=&app_name=AF-Renderer::Self&_=1667950947043&sid=a641aaa3010252386ee44cf3d15ec734
```
but once used it seems to expire, here is what it returned for me 

```json
{
    "status": "done",
    "data": "<?xml version=\"1.0\"?><Responses><RequestResponse><DatabaseResponse><Fields><Field Type=\"varchar\" Name=\"echo_service_available\"\/><Field Type=\"varchar\" Name=\"echo_service_error_message\"\/><Field Type=\"varchar\" Name=\"echo_waste_services_available_at_address\"\/><Field Type=\"varchar\" Name=\"echo_address_is_residential\"\/><Field Type=\"varchar\" Name=\"echo_land_use_code\"\/><Field Type=\"varchar\" Name=\"echo_refuse_service_available\"\/><Field Type=\"varchar\" Name=\"echo_refuse_service_id\"\/><Field Type=\"varchar\" Name=\"echo_refuse_service_name\"\/><Field Type=\"varchar\" Name=\"echo_refuse_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_refuse_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_refuse_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_refuse_last_date\"\/><Field Type=\"varchar\" Name=\"echo_refuse_next_date\"\/><Field Type=\"varchar\" Name=\"echo_refuse_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_refuse_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_service_available\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_service_id\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_service_name\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_last_date\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_next_date\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_service_available\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_service_id\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_service_name\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_last_date\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_next_date\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_service_available\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_service_id\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_service_name\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_last_date\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_next_date\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_service_available\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_recycling_service_id\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_service_name\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_recycling_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_recycling_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_last_date\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_next_date\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_waste_calendar_link\"\/><Field Type=\"varchar\" Name=\"echo_call_status\"\/><Field Type=\"varchar\" Name=\"echo_error_message\"\/><\/Fields><Rows><Row id=\"0\"><result column=\"echo_service_available\" isNull=\"False\">True<\/result><result column=\"echo_service_error_message\" isNull=\"True\"\/><result column=\"echo_waste_services_available_at_address\" isNull=\"False\">True<\/result><result column=\"echo_address_is_residential\" isNull=\"False\">True<\/result><result column=\"echo_land_use_code\" isNull=\"False\">RD04<\/result><result column=\"echo_refuse_service_available\" isNull=\"False\">True<\/result><result column=\"echo_refuse_service_id\" isNull=\"False\">355<\/result><result column=\"echo_refuse_service_name\" isNull=\"False\">Domestic Refuse Collection<\/result><result column=\"echo_refuse_task_type_id\" isNull=\"False\">1903<\/result><result column=\"echo_refuse_task_type_name\" isNull=\"False\">Collect Refuse Bin<\/result><result column=\"echo_refuse_schedule_description\" isNull=\"False\">Thursday every other week<\/result><result column=\"echo_refuse_last_date\" isNull=\"False\">2022-11-03 06:45:00<\/result><result column=\"echo_refuse_next_date\" isNull=\"False\">2022-11-17 00:00:00<\/result><result column=\"echo_refuse_subscription_end_date\" isNull=\"True\"\/><result column=\"echo_refuse_asset_type_name\" isNull=\"True\"\/><result column=\"echo_food_waste_service_available\" isNull=\"False\">True<\/result><result column=\"echo_food_waste_service_id\" isNull=\"False\">357<\/result><result column=\"echo_food_waste_service_name\" isNull=\"False\">Domestic Recycling Collection<\/result><result column=\"echo_food_waste_task_type_id\" isNull=\"False\">1908<\/result><result column=\"echo_food_waste_task_type_name\" isNull=\"False\">Collect Food Caddy<\/result><result column=\"echo_food_waste_schedule_description\" isNull=\"False\">Thursday every week<\/result><result column=\"echo_food_waste_last_date\" isNull=\"False\">2022-11-03 06:45:00<\/result><result column=\"echo_food_waste_next_date\" isNull=\"False\">2022-11-10 00:00:00<\/result><result column=\"echo_food_waste_subscription_end_date\" isNull=\"True\"\/><result column=\"echo_food_waste_asset_type_name\" isNull=\"True\"\/><result column=\"echo_paper_and_card_service_available\" isNull=\"False\">True<\/result><result column=\"echo_paper_and_card_service_id\" isNull=\"False\">357<\/result><result column=\"echo_paper_and_card_service_name\" isNull=\"False\">Domestic Recycling Collection<\/result><result column=\"echo_paper_and_card_task_type_id\" isNull=\"False\">1906<\/result><result column=\"echo_paper_and_card_task_type_name\" isNull=\"False\">Collect Paper &amp; Card Recycling Bin<\/result><result column=\"echo_paper_and_card_schedule_description\" isNull=\"False\">Thursday every other week<\/result><result column=\"echo_paper_and_card_last_date\" isNull=\"False\">2022-11-03 06:45:00<\/result><result column=\"echo_paper_and_card_next_date\" isNull=\"False\">2022-11-17 00:00:00<\/result><result column=\"echo_paper_and_card_subscription_end_date\" isNull=\"True\"\/><result column=\"echo_paper_and_card_asset_type_name\" isNull=\"True\"\/><result column=\"echo_mixed_recycling_service_available\" isNull=\"False\">True<\/result><result column=\"echo_mixed_recycling_service_id\" isNull=\"False\">357<\/result><result column=\"echo_mixed_recycling_service_name\" isNull=\"False\">Domestic Recycling Collection<\/result><result column=\"echo_mixed_recycling_task_type_id\" isNull=\"False\">1909<\/result><result column=\"echo_mixed_recycling_task_type_name\" isNull=\"False\">Collect Mixed Recycling Box<\/result><result column=\"echo_mixed_recycling_schedule_description\" isNull=\"False\">Thursday every other week<\/result><result column=\"echo_mixed_recycling_last_date\" isNull=\"False\">2022-10-27 06:45:00<\/result><result column=\"echo_mixed_recycling_next_date\" isNull=\"False\">2022-11-10 00:00:00<\/result><result column=\"echo_mixed_recycling_subscription_end_date\" isNull=\"True\"\/><result column=\"echo_mixed_recycling_asset_type_name\" isNull=\"True\"\/><result column=\"echo_garden_waste_service_available\" isNull=\"False\">True<\/result><result column=\"echo_garden_waste_recycling_service_id\" isNull=\"False\">409<\/result><result column=\"echo_garden_waste_service_name\" isNull=\"False\">Domestic Green Collection<\/result><result column=\"echo_garden_waste_recycling_task_type_id\" isNull=\"False\">2247<\/result><result column=\"echo_garden_waste_recycling_task_type_name\" isNull=\"False\">Collect Domestic Green<\/result><result column=\"echo_garden_waste_schedule_description\" isNull=\"False\">Thursday every other week<\/result><result column=\"echo_garden_waste_last_date\" isNull=\"False\">2022-11-03 06:00:00<\/result><result column=\"echo_garden_waste_next_date\" isNull=\"False\">2022-11-17 00:00:00<\/result><result column=\"echo_garden_waste_subscription_end_date\" isNull=\"False\">2023-03-31 00:00:00<\/result><result column=\"echo_garden_waste_asset_type_name\" isNull=\"False\">Garden 240L<\/result><result column=\"echo_waste_calendar_link\" isNull=\"False\">https:\/\/www.kingston.gov.uk\/downloads\/file\/1602\/kingston-kerbside-x-10-kinds-thursday-a<\/result><result column=\"echo_call_status\" isNull=\"False\">True<\/result><result column=\"echo_error_message\" isNull=\"True\"\/><\/Row><\/Rows><\/DatabaseResponse><\/RequestResponse><\/Responses>",
    "integration": {
        "id": "601a61f9a3188",
        "type": "http",
        "transformed": {
            "error": null,
            "xml_data": "<?xml version=\"1.0\"?><Responses><RequestResponse><DatabaseResponse><Fields><Field Type=\"varchar\" Name=\"echo_service_available\"\/><Field Type=\"varchar\" Name=\"echo_service_error_message\"\/><Field Type=\"varchar\" Name=\"echo_waste_services_available_at_address\"\/><Field Type=\"varchar\" Name=\"echo_address_is_residential\"\/><Field Type=\"varchar\" Name=\"echo_land_use_code\"\/><Field Type=\"varchar\" Name=\"echo_refuse_service_available\"\/><Field Type=\"varchar\" Name=\"echo_refuse_service_id\"\/><Field Type=\"varchar\" Name=\"echo_refuse_service_name\"\/><Field Type=\"varchar\" Name=\"echo_refuse_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_refuse_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_refuse_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_refuse_last_date\"\/><Field Type=\"varchar\" Name=\"echo_refuse_next_date\"\/><Field Type=\"varchar\" Name=\"echo_refuse_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_refuse_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_service_available\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_service_id\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_service_name\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_last_date\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_next_date\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_food_waste_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_service_available\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_service_id\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_service_name\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_last_date\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_next_date\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_paper_and_card_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_service_available\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_service_id\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_service_name\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_last_date\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_next_date\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_mixed_recycling_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_service_available\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_recycling_service_id\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_service_name\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_recycling_task_type_id\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_recycling_task_type_name\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_schedule_description\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_last_date\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_next_date\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_subscription_end_date\"\/><Field Type=\"varchar\" Name=\"echo_garden_waste_asset_type_name\"\/><Field Type=\"varchar\" Name=\"echo_waste_calendar_link\"\/><Field Type=\"varchar\" Name=\"echo_call_status\"\/><Field Type=\"varchar\" Name=\"echo_error_message\"\/><\/Fields><Rows><Row id=\"0\"><result column=\"echo_service_available\" isNull=\"False\">True<\/result><result column=\"echo_service_error_message\" isNull=\"True\"\/><result column=\"echo_waste_services_available_at_address\" isNull=\"False\">True<\/result><result column=\"echo_address_is_residential\" isNull=\"False\">True<\/result><result column=\"echo_land_use_code\" isNull=\"False\">RD04<\/result><result column=\"echo_refuse_service_available\" isNull=\"False\">True<\/result><result column=\"echo_refuse_service_id\" isNull=\"False\">355<\/result><result column=\"echo_refuse_service_name\" isNull=\"False\">Domestic Refuse Collection<\/result><result column=\"echo_refuse_task_type_id\" isNull=\"False\">1903<\/result><result column=\"echo_refuse_task_type_name\" isNull=\"False\">Collect Refuse Bin<\/result><result column=\"echo_refuse_schedule_description\" isNull=\"False\">Thursday every other week<\/result><result column=\"echo_refuse_last_date\" isNull=\"False\">2022-11-03 06:45:00<\/result><result column=\"echo_refuse_next_date\" isNull=\"False\">2022-11-17 00:00:00<\/result><result column=\"echo_refuse_subscription_end_date\" isNull=\"True\"\/><result column=\"echo_refuse_asset_type_name\" isNull=\"True\"\/><result column=\"echo_food_waste_service_available\" isNull=\"False\">True<\/result><result column=\"echo_food_waste_service_id\" isNull=\"False\">357<\/result><result column=\"echo_food_waste_service_name\" isNull=\"False\">Domestic Recycling Collection<\/result><result column=\"echo_food_waste_task_type_id\" isNull=\"False\">1908<\/result><result column=\"echo_food_waste_task_type_name\" isNull=\"False\">Collect Food Caddy<\/result><result column=\"echo_food_waste_schedule_description\" isNull=\"False\">Thursday every week<\/result><result column=\"echo_food_waste_last_date\" isNull=\"False\">2022-11-03 06:45:00<\/result><result column=\"echo_food_waste_next_date\" isNull=\"False\">2022-11-10 00:00:00<\/result><result column=\"echo_food_waste_subscription_end_date\" isNull=\"True\"\/><result column=\"echo_food_waste_asset_type_name\" isNull=\"True\"\/><result column=\"echo_paper_and_card_service_available\" isNull=\"False\">True<\/result><result column=\"echo_paper_and_card_service_id\" isNull=\"False\">357<\/result><result column=\"echo_paper_and_card_service_name\" isNull=\"False\">Domestic Recycling Collection<\/result><result column=\"echo_paper_and_card_task_type_id\" isNull=\"False\">1906<\/result><result column=\"echo_paper_and_card_task_type_name\" isNull=\"False\">Collect Paper &amp; Card Recycling Bin<\/result><result column=\"echo_paper_and_card_schedule_description\" isNull=\"False\">Thursday every other week<\/result><result column=\"echo_paper_and_card_last_date\" isNull=\"False\">2022-11-03 06:45:00<\/result><result column=\"echo_paper_and_card_next_date\" isNull=\"False\">2022-11-17 00:00:00<\/result><result column=\"echo_paper_and_card_subscription_end_date\" isNull=\"True\"\/><result column=\"echo_paper_and_card_asset_type_name\" isNull=\"True\"\/><result column=\"echo_mixed_recycling_service_available\" isNull=\"False\">True<\/result><result column=\"echo_mixed_recycling_service_id\" isNull=\"False\">357<\/result><result column=\"echo_mixed_recycling_service_name\" isNull=\"False\">Domestic Recycling Collection<\/result><result column=\"echo_mixed_recycling_task_type_id\" isNull=\"False\">1909<\/result><result column=\"echo_mixed_recycling_task_type_name\" isNull=\"False\">Collect Mixed Recycling Box<\/result><result column=\"echo_mixed_recycling_schedule_description\" isNull=\"False\">Thursday every other week<\/result><result column=\"echo_mixed_recycling_last_date\" isNull=\"False\">2022-10-27 06:45:00<\/result><result column=\"echo_mixed_recycling_next_date\" isNull=\"False\">2022-11-10 00:00:00<\/result><result column=\"echo_mixed_recycling_subscription_end_date\" isNull=\"True\"\/><result column=\"echo_mixed_recycling_asset_type_name\" isNull=\"True\"\/><result column=\"echo_garden_waste_service_available\" isNull=\"False\">True<\/result><result column=\"echo_garden_waste_recycling_service_id\" isNull=\"False\">409<\/result><result column=\"echo_garden_waste_service_name\" isNull=\"False\">Domestic Green Collection<\/result><result column=\"echo_garden_waste_recycling_task_type_id\" isNull=\"False\">2247<\/result><result column=\"echo_garden_waste_recycling_task_type_name\" isNull=\"False\">Collect Domestic Green<\/result><result column=\"echo_garden_waste_schedule_description\" isNull=\"False\">Thursday every other week<\/result><result column=\"echo_garden_waste_last_date\" isNull=\"False\">2022-11-03 06:00:00<\/result><result column=\"echo_garden_waste_next_date\" isNull=\"False\">2022-11-17 00:00:00<\/result><result column=\"echo_garden_waste_subscription_end_date\" isNull=\"False\">2023-03-31 00:00:00<\/result><result column=\"echo_garden_waste_asset_type_name\" isNull=\"False\">Garden 240L<\/result><result column=\"echo_waste_calendar_link\" isNull=\"False\">https:\/\/www.kingston.gov.uk\/downloads\/file\/1602\/kingston-kerbside-x-10-kinds-thursday-a<\/result><result column=\"echo_call_status\" isNull=\"False\">True<\/result><result column=\"echo_error_message\" isNull=\"True\"\/><\/Row><\/Rows><\/DatabaseResponse><\/RequestResponse><\/Responses>",
            "rows_data": {
                "0": {
                    "echo_service_available": "True",
                    "echo_service_error_message": "",
                    "echo_waste_services_available_at_address": "True",
                    "echo_address_is_residential": "True",
                    "echo_land_use_code": "RD04",
                    "echo_refuse_service_available": "True",
                    "echo_refuse_service_id": "355",
                    "echo_refuse_service_name": "Domestic Refuse Collection",
                    "echo_refuse_task_type_id": "1903",
                    "echo_refuse_task_type_name": "Collect Refuse Bin",
                    "echo_refuse_schedule_description": "Thursday every other week",
                    "echo_refuse_last_date": "2022-11-03 06:45:00",
                    "echo_refuse_next_date": "2022-11-17 00:00:00",
                    "echo_refuse_subscription_end_date": "",
                    "echo_refuse_asset_type_name": "",
                    "echo_food_waste_service_available": "True",
                    "echo_food_waste_service_id": "357",
                    "echo_food_waste_service_name": "Domestic Recycling Collection",
                    "echo_food_waste_task_type_id": "1908",
                    "echo_food_waste_task_type_name": "Collect Food Caddy",
                    "echo_food_waste_schedule_description": "Thursday every week",
                    "echo_food_waste_last_date": "2022-11-03 06:45:00",
                    "echo_food_waste_next_date": "2022-11-10 00:00:00",
                    "echo_food_waste_subscription_end_date": "",
                    "echo_food_waste_asset_type_name": "",
                    "echo_paper_and_card_service_available": "True",
                    "echo_paper_and_card_service_id": "357",
                    "echo_paper_and_card_service_name": "Domestic Recycling Collection",
                    "echo_paper_and_card_task_type_id": "1906",
                    "echo_paper_and_card_task_type_name": "Collect Paper & Card Recycling Bin",
                    "echo_paper_and_card_schedule_description": "Thursday every other week",
                    "echo_paper_and_card_last_date": "2022-11-03 06:45:00",
                    "echo_paper_and_card_next_date": "2022-11-17 00:00:00",
                    "echo_paper_and_card_subscription_end_date": "",
                    "echo_paper_and_card_asset_type_name": "",
                    "echo_mixed_recycling_service_available": "True",
                    "echo_mixed_recycling_service_id": "357",
                    "echo_mixed_recycling_service_name": "Domestic Recycling Collection",
                    "echo_mixed_recycling_task_type_id": "1909",
                    "echo_mixed_recycling_task_type_name": "Collect Mixed Recycling Box",
                    "echo_mixed_recycling_schedule_description": "Thursday every other week",
                    "echo_mixed_recycling_last_date": "2022-10-27 06:45:00",
                    "echo_mixed_recycling_next_date": "2022-11-10 00:00:00",
                    "echo_mixed_recycling_subscription_end_date": "",
                    "echo_mixed_recycling_asset_type_name": "",
                    "echo_garden_waste_service_available": "True",
                    "echo_garden_waste_recycling_service_id": "409",
                    "echo_garden_waste_service_name": "Domestic Green Collection",
                    "echo_garden_waste_recycling_task_type_id": "2247",
                    "echo_garden_waste_recycling_task_type_name": "Collect Domestic Green",
                    "echo_garden_waste_schedule_description": "Thursday every other week",
                    "echo_garden_waste_last_date": "2022-11-03 06:00:00",
                    "echo_garden_waste_next_date": "2022-11-17 00:00:00",
                    "echo_garden_waste_subscription_end_date": "2023-03-31 00:00:00",
                    "echo_garden_waste_asset_type_name": "Garden 240L",
                    "echo_waste_calendar_link": "https:\/\/www.kingston.gov.uk\/downloads\/file\/1602\/kingston-kerbside-x-10-kinds-thursday-a",
                    "echo_call_status": "True",
                    "echo_error_message": ""
                }
            },
            "fields_data": {
                "echo_service_available": {
                    "type": "varchar",
                    "name": "echo_service_available"
                },
                "echo_service_error_message": {
                    "type": "varchar",
                    "name": "echo_service_error_message"
                },
                "echo_waste_services_available_at_address": {
                    "type": "varchar",
                    "name": "echo_waste_services_available_at_address"
                },
                "echo_address_is_residential": {
                    "type": "varchar",
                    "name": "echo_address_is_residential"
                },
                "echo_land_use_code": {
                    "type": "varchar",
                    "name": "echo_land_use_code"
                },
                "echo_refuse_service_available": {
                    "type": "varchar",
                    "name": "echo_refuse_service_available"
                },
                "echo_refuse_service_id": {
                    "type": "varchar",
                    "name": "echo_refuse_service_id"
                },
                "echo_refuse_service_name": {
                    "type": "varchar",
                    "name": "echo_refuse_service_name"
                },
                "echo_refuse_task_type_id": {
                    "type": "varchar",
                    "name": "echo_refuse_task_type_id"
                },
                "echo_refuse_task_type_name": {
                    "type": "varchar",
                    "name": "echo_refuse_task_type_name"
                },
                "echo_refuse_schedule_description": {
                    "type": "varchar",
                    "name": "echo_refuse_schedule_description"
                },
                "echo_refuse_last_date": {
                    "type": "varchar",
                    "name": "echo_refuse_last_date"
                },
                "echo_refuse_next_date": {
                    "type": "varchar",
                    "name": "echo_refuse_next_date"
                },
                "echo_refuse_subscription_end_date": {
                    "type": "varchar",
                    "name": "echo_refuse_subscription_end_date"
                },
                "echo_refuse_asset_type_name": {
                    "type": "varchar",
                    "name": "echo_refuse_asset_type_name"
                },
                "echo_food_waste_service_available": {
                    "type": "varchar",
                    "name": "echo_food_waste_service_available"
                },
                "echo_food_waste_service_id": {
                    "type": "varchar",
                    "name": "echo_food_waste_service_id"
                },
                "echo_food_waste_service_name": {
                    "type": "varchar",
                    "name": "echo_food_waste_service_name"
                },
                "echo_food_waste_task_type_id": {
                    "type": "varchar",
                    "name": "echo_food_waste_task_type_id"
                },
                "echo_food_waste_task_type_name": {
                    "type": "varchar",
                    "name": "echo_food_waste_task_type_name"
                },
                "echo_food_waste_schedule_description": {
                    "type": "varchar",
                    "name": "echo_food_waste_schedule_description"
                },
                "echo_food_waste_last_date": {
                    "type": "varchar",
                    "name": "echo_food_waste_last_date"
                },
                "echo_food_waste_next_date": {
                    "type": "varchar",
                    "name": "echo_food_waste_next_date"
                },
                "echo_food_waste_subscription_end_date": {
                    "type": "varchar",
                    "name": "echo_food_waste_subscription_end_date"
                },
                "echo_food_waste_asset_type_name": {
                    "type": "varchar",
                    "name": "echo_food_waste_asset_type_name"
                },
                "echo_paper_and_card_service_available": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_service_available"
                },
                "echo_paper_and_card_service_id": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_service_id"
                },
                "echo_paper_and_card_service_name": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_service_name"
                },
                "echo_paper_and_card_task_type_id": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_task_type_id"
                },
                "echo_paper_and_card_task_type_name": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_task_type_name"
                },
                "echo_paper_and_card_schedule_description": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_schedule_description"
                },
                "echo_paper_and_card_last_date": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_last_date"
                },
                "echo_paper_and_card_next_date": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_next_date"
                },
                "echo_paper_and_card_subscription_end_date": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_subscription_end_date"
                },
                "echo_paper_and_card_asset_type_name": {
                    "type": "varchar",
                    "name": "echo_paper_and_card_asset_type_name"
                },
                "echo_mixed_recycling_service_available": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_service_available"
                },
                "echo_mixed_recycling_service_id": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_service_id"
                },
                "echo_mixed_recycling_service_name": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_service_name"
                },
                "echo_mixed_recycling_task_type_id": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_task_type_id"
                },
                "echo_mixed_recycling_task_type_name": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_task_type_name"
                },
                "echo_mixed_recycling_schedule_description": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_schedule_description"
                },
                "echo_mixed_recycling_last_date": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_last_date"
                },
                "echo_mixed_recycling_next_date": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_next_date"
                },
                "echo_mixed_recycling_subscription_end_date": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_subscription_end_date"
                },
                "echo_mixed_recycling_asset_type_name": {
                    "type": "varchar",
                    "name": "echo_mixed_recycling_asset_type_name"
                },
                "echo_garden_waste_service_available": {
                    "type": "varchar",
                    "name": "echo_garden_waste_service_available"
                },
                "echo_garden_waste_recycling_service_id": {
                    "type": "varchar",
                    "name": "echo_garden_waste_recycling_service_id"
                },
                "echo_garden_waste_service_name": {
                    "type": "varchar",
                    "name": "echo_garden_waste_service_name"
                },
                "echo_garden_waste_recycling_task_type_id": {
                    "type": "varchar",
                    "name": "echo_garden_waste_recycling_task_type_id"
                },
                "echo_garden_waste_recycling_task_type_name": {
                    "type": "varchar",
                    "name": "echo_garden_waste_recycling_task_type_name"
                },
                "echo_garden_waste_schedule_description": {
                    "type": "varchar",
                    "name": "echo_garden_waste_schedule_description"
                },
                "echo_garden_waste_last_date": {
                    "type": "varchar",
                    "name": "echo_garden_waste_last_date"
                },
                "echo_garden_waste_next_date": {
                    "type": "varchar",
                    "name": "echo_garden_waste_next_date"
                },
                "echo_garden_waste_subscription_end_date": {
                    "type": "varchar",
                    "name": "echo_garden_waste_subscription_end_date"
                },
                "echo_garden_waste_asset_type_name": {
                    "type": "varchar",
                    "name": "echo_garden_waste_asset_type_name"
                },
                "echo_waste_calendar_link": {
                    "type": "varchar",
                    "name": "echo_waste_calendar_link"
                },
                "echo_call_status": {
                    "type": "varchar",
                    "name": "echo_call_status"
                },
                "echo_error_message": {
                    "type": "varchar",
                    "name": "echo_error_message"
                }
            },
            "select_data": [
                {
                    "label": "",
                    "value": ""
                }
            ],
            "results_page": null,
            "post_transform": null,
            "pre_transform": null
        },
        "tokens": [
            "{middleman_api_root_uri}",
            "{echo_borough}",
            "{echo_uprn}",
            "{echo_suppress_errors}"
        ],
        "runtimes": {
            "integration": {
                "template": "14ms",
                "lim": "13ms",
                "http": "3481ms"
            },
            "api": "3481ms",
            "lim": "13ms"
        },
        "log_id": "6ae09505-0531-47d1-91c4-716665f438a0"
    },
    "hash": "10C0B1122B11C7414F14F693A67AF4F949733BD87821E10EDA3EA995D2705ED7"
}
```
It looks like if we can crack the initial request, the data comes back in a form that should be fairly easy to read in the collection dates and types

**Comment by DevilasNZ** [comment]
I had a similar issue with my region, that does make it a bit harder unfortunately. It looks like there's a session ID that gets dished out, see the "&sid=a641aaa3010252386ee44cf3d15ec734" part at the end of the URL. If you can find which request returns that session ID hopefully that will help

**Comment by skipishere** [comment]
The session id part seems fairly easy, I did find the request that returned the session id, the one I really struggled with was finding what call generated the `?id=601a61f9a3188` as the page calls the `https://kingston-self.achieveservice.com/apibroker/runLookup` end point a number of times and the id seems to return different data sets (such as who the councillor for the area is)

**Comment by ximon** [comment]
Take a look at PR 472, this seems very similar (both use AcheiveForms)

**Comment by ximon** [comment]
I had some spare time so I've made the changes to work for kingston council and submitted PR 483 for it.

**Comment by 5ila5** [comment]
Can be closed fixer by #483
